### PR TITLE
fix(core): pass all data to customHTML for scatter-stacked charts

### PR DESCRIPTION
**Pull request by obambrough**
Provide all data to the customHTML tooltip callback for scatter-stacked, rather than just the group
name, domain and range values.

### Demo screenshot or recording
Before:
image

After (with additional properties from data series):
image

### Review checklist (for reviewers only)
- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Meets the [code style guide](https://github.com/carbon-design-system/carbon-charts/wiki/Code-style-guide)
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)

### CODE REVIEW

File: packages/core/src/components/graphs/scatter-stacked.ts [hunk 1]
```diff
@@ -121,6 +121,17 @@ export class StackedScatter extends Scatter {
 				}
 			});
 		});
-		return tooltipData;
+
+		return this.model.getDisplayData().filter((d) => {
+			return (
+				tooltipData.find((ttd) => {
```
Comment (theiliad) [inline packages/core/src/components/graphs/scatter-stacked.ts:127]:
```suggestion
				tooltipData.find((tooltipDatapoint) => {
```

File: packages/core/src/components/graphs/scatter-stacked.ts [hunk 2]
```diff
@@ -121,6 +121,17 @@ export class StackedScatter extends Scatter {
 				}
 			});
 		});
-		return tooltipData;
+
+		return this.model.getDisplayData().filter((d) => {
```
Comment (theiliad) [inline packages/core/src/components/graphs/scatter-stacked.ts:125]:
```suggestion
		return this.model.getDisplayData().filter((datapoint) => {
```

File: packages/core/src/components/graphs/scatter-stacked.ts [hunk 3]
```diff
@@ -121,6 +121,17 @@ export class StackedScatter extends Scatter {
...
+			return (
+				tooltipData.find((ttd) => {
+					return (
+						ttd[groupMapsTo] == d[groupMapsTo] &&
+						ttd[domainIdentifier] == d[domainIdentifier] &&
+						ttd[rangeIdentifier] == d[rangeIdentifier]
```
Comment (theiliad) [inline packages/core/src/components/graphs/scatter-stacked.ts:131]: and updating the same names here. I think this'll make it a bit more readable

**Review by theiliad — CHANGES_REQUESTED** [review]

File: packages/core/src/components/graphs/scatter-stacked.ts (hunk 3 above)
Comment (obambrough) [reply packages/core/src/components/graphs/scatter-stacked.ts]: I've pushed the changes with the updated names

**Review by theiliad — APPROVED** [review]

**Review by sophiiae — APPROVED** [review]

**Review by natashadecoste — APPROVED** [review]

**Pull request merged**